examples/shader/post_processing.rs

```diff
@@ -0,0 +1,340 @@
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 10.0)),
+        ..default()
+    });
+
+    // First pass camera
+    let render_target = RenderTarget::Image(image_handle.clone());
+    clear_colors.insert(render_target.clone(), Color::WHITE);
+    commands
+        .spawn_bundle(PerspectiveCameraBundle::<FirstPassCamera> {
+            camera: Camera {
+                target: render_target,
+                ..default()
+            },
+            transform: Transform::from_translation(Vec3::new(0.0, 0.0, 15.0))
+                .looking_at(Vec3::default(), Vec3::Y),
+            ..PerspectiveCameraBundle::new()
+        })
+        .insert(first_pass_layer);
+    // NOTE: omitting the RenderLayers component for this camera may cause a validation error:
+    //
+    // thread 'main' panicked at 'wgpu error: Validation Error
+    //
+    //    Caused by:
+    //        In a RenderPass
+    //          note: encoder = `<CommandBuffer-(0, 1, Metal)>`
+    //        In a pass parameter
+    //          note: command buffer = `<CommandBuffer-(0, 1, Metal)>`
+    //        Attempted to use texture (5, 1, Metal) mips 0..1 layers 0..1 as a combination of COLOR_TARGET within a usage scope.
+    //
+    // This happens because the texture would be written and read in the same frame, which is not allowed.
+    // So either render layers must be used to avoid this, or the texture must be double buffered.
+
+    let quad_size = 400.0;
+    let quad_handle = meshes.add(Mesh::from(shape::Quad::new(Vec2::splat(quad_size))));
+
+    // This material has the texture that has been rendered.
+    let material_handle = post_processing_materials.add(PostProcessingMaterial {
+        source_image: image_handle,
+    });
+
+    // Main pass 2d quad, with material containing the rendered first pass texture, with a custom shader.
+    commands.spawn_bundle(MaterialMesh2dBundle {
+        mesh: quad_handle.into(),
+        material: material_handle,
+        transform: Transform {
+            translation: Vec3::new(0.0, 0.0, 1.5),
+            rotation: Quat::from_rotation_x(-std::f32::consts::PI / 5.0),
+            ..default()
+        },
+        ..default()
+    });
+
+    // The main pass camera.
+    let mut camera = OrthographicCameraBundle::new_2d();
+    camera.orthographic_projection.scale = 0.10f32;
+    camera.transform.translation = Vec3::new(0f32, 100f32, -1f32);
+    commands.spawn_bundle(camera);
+}
+
+/// Rotates the inner cube (first pass)
+fn first_pass_cube_rotator_system(
+    time: Res<Time>,
+    mut query: Query<&mut Transform, With<FirstPassCube>>,
+) {
+    for mut transform in query.iter_mut() {
+        transform.rotation *= Quat::from_rotation_x(1.5 * time.delta_seconds());
+        transform.rotation *= Quat::from_rotation_z(1.3 * time.delta_seconds());
+    }
+}
+
+///! Declaration of the custom material handling post processing effect
```

Comment:
Not sure how that doc comment will render.

Reply:
it won't, doc comment on examples are not rendered

also, it's `///` to doc comment the item following, or `//!` to doc comment the item containing

Reply:
Yeah I wanted to document the whole area underneath, I'll work on that